nav2_controller/src/controller_server.cpp

```diff
@@ -384,10 +384,16 @@ void ControllerServer::computeControl()
         return;
       }
 
-      // Don't compute a trajectory until costmap is valid (after clear costmap)
-      rclcpp::Rate r(100);
-      while (!costmap_ros_->isCurrent()) {
-        r.sleep();
+      if (costmap_ros_->isUpdateOnRequest()){
+        RCLCPP_DEBUG(get_logger(), "Updating costmap on request!");
+        costmap_ros_->updateMapOnRequest();
+      }
+      else {
+        // Don't compute a trajectory until costmap is valid (after clear costmap)
+        rclcpp::Rate r(100);
+        while (!costmap_ros_->isCurrent()) {
+          r.sleep();
+        }
```

Comment:
We need to check if we are updating on request (that is manually), mean we have no updating loop thread enabled. Thus, in this case I believe we need to 1. update map manually by calling new API and then 2. wait until costmap is being updated. But I see these two steps are in separate if-branches. Why this logic was introduced here?